gnatsd/check.py

```diff
@@ -0,0 +1,150 @@
+# (C) Datadog, Inc. 2010-2016
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+import requests
+
+# stdlib
+
+# project
+from checks import AgentCheck
+
+EVENT_TYPE = SOURCE_TYPE_NAME = 'gnatsd'
+
+class GnatsdCheck(AgentCheck):
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
+        self.counts = {}
+
+    def check(self, instance):
+        self.InstanceCheck(instance, self).check()
+
+    class InstanceCheck:
+        SERVICE_CHECK_NAME = 'gnatsd.can_connect'
+
+        METRICS = {
+            'varz': {
+                'connections': 'gauge',
+                'subscriptions': 'gauge',
+                'slow_consumers': 'count',
+                'remotes': 'gauge',
+                'routes': 'gauge',
+                'in_msgs': 'count',
+                'out_msgs': 'count',
+                'in_bytes': 'count',
+                'out_bytes': 'count',
+                'mem': 'gauge'
+            },
+            'connz': {
+                'num_connections': 'gauge',
+                'total': 'count',
+                'connections': {
+                    'pending_bytes': 'gauge',
+                    'in_msgs': 'count',
+                    'out_msgs': 'count',
+                    'subscriptions': 'gauge',
+                    'in_bytes': 'count',
+                    'out_bytes': 'count'
+                }
+            },
+            'routez': {
+                'num_routes': 'gauge',
+                'routes': {
+                    'pending_size': 'gauge',
+                    'in_msgs': 'count',
+                    'out_msgs': 'count',
+                    'subscriptions': 'gauge',
+                    'in_bytes': 'count',
+                    'out_bytes': 'count'
+                }
+            }
+        }
+
+        TAGS = {
+            'varz': ['server_id'],
+            'connz.connections': ['cid','ip','name','lang','version'],
+            'routez.routes': ['rid','remote_id','ip']
+        }
+
+        def __init__(self, instance, checker):
+            self.instance = instance
+            self.checker = checker
+            self.config = self.Config(instance)
```

Comment:
`Config` seems like it could be replaced with a `dict`.

Reply:
Nah - it reads in the instance and then returns calculated values.  Otherwise we'd have to do those calculations in human heads.  If the values were constants I'd agree with you for sure.

Reply:
You could also do this:

```python
{
    'port': int(instance.get('port', 8222))
}
```